qiskit_nature/results/bopes_sampler_result.py

```diff
@@ -44,9 +44,14 @@ def points(self) -> List[float]:
         return self._points
 
     @property
-    def energies(self) -> List[float]:
+    def energies(self) -> Union[List[float], List[List[float]]]:
         """returns list of energies."""
-        return self._energies
+        formatted_energies: Union[List[float], List[List[float]]]
+        if len(self._energies[0]) == 1:
+            formatted_energies = [self._energies[k][0] for k in range(len(self._energies))]
+        else:
+            formatted_energies = self._energies
```

Comment:
Is this logic correct? It just seems odd in that if the len of the element is 1 we take out the only element and make a flat list of it, so self._energies is always a list of lists but of the size of the inner list is 1 then we make it a list of floats. This is simply because of how you format up self._energies from the result? Maybe I am confusing things in my mind with the excited state result from the eigensolver since as I recall that was a list of tuples, each tuple being (value, variance) pair

Reply:
I did some digging. What is being done here is the differentiation between the ground- and excited-states case.
In both cases, `self._energies` will be a list of numpy arrays. In the case of the ground-state, all of these arrays have length 1 in which case this list is being flattened. In the excited-states case, these arrays are longer and, thus, are left untouched.

I think this is fine as is, if a comment gets added, explaining this behavior, similar to as I did here just now

Reply:
Indeed, when only `GroundStateSolver` were used, BOPES was flattening the result before giving it to the BOPESResult. Now that it also accepts `ExcitedStatesSolver` for which the output is a list of (lists of size N, with N the Nb of excited states), it is more natural to keep the list of (lists of size 1) and to only flatten it later in `BOPESResult`.

Reply:
I'm actually wondering now, if this flattening is needed at all :thinking:
If a user were to use the resulting `BOPESSamplerResult` object, they would then again also need to differentiate the case. If we were to leave this as a list of numpy arrays, it would always behave the same

Reply:
Regarding your comment on the flattening of the BOPESResult energies, it is only a matter of backward compatibility. All the tests and tutorials use List(float) instead of List(List(float)). I wanted to keep it transparent for GroundStateSolver and add the option for ExcitedStatesSolver